packages/core/lib/segments/attributes/subsegment.js

```diff
@@ -38,10 +39,24 @@ Subsegment.prototype.init = function init(name) {
 
 Subsegment.prototype.addNewSubsegment = function addNewSubsegment(name) {
   var subsegment = new Subsegment(name);
+  subsegment.isSampled = true;
```

Comment:
Great callout! I believe checking the value of `subsegment.segment.notTraced` would be more accurate in this case as to capture the parent _segment's_ sampling decision not the parent _subsegment_ (since it is possible to have subsegments of subsegments.